### PR TITLE
Update limit manager role permissions ADR

### PR DESCRIPTION
**Pull request by nikola-jokic**
We extended permissions to patch the role in order to do a cleanup on `helm uninstall`

ADR has been updated to reflect that change, and enhance it with annotations required for controller to do a proper cleanup.

### CODE REVIEW

File: docs/adrs/2022-12-05-adding-labels-k8s-resources.md [hunk 1]
```diff
@@ -86,4 +86,4 @@ Or for example if they're having problems specifically with runners:
 This way users don't have to understand ARC moving parts but we still have a
 way to target them specifically if we need to.
 
-[^1]: Superseded by [ADR 2023-04-14](2023-04-14-adding-labels-k8s-resources.md)
```
Comment (TingluoHuang) [inline docs/adrs/2022-12-05-adding-labels-k8s-resources.md:89]: welcome back from the future. 😄

File: docs/adrs/2023-04-11-limit-manager-role-permission.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
+# ADR 2023-02-10: Limit Permissions for Service Accounts in Actions-Runner-Controller
```
Comment (TingluoHuang) [inline docs/adrs/2023-04-11-limit-manager-role-permission.md]:
```suggestion
# ADR 2023-04-11: Limit Permissions for Service Accounts in Actions-Runner-Controller
```

File: docs/adrs/2023-04-11-limit-manager-role-permission.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+To help these customers and improve security for `actions-runner-controller` in general, we will try to limit the `ClusterRole` permission of the controller manager's service account down to the following:
+
+- Get/List/Create/Delete/Update/Patch/Watch on `AutoScalingRunnerSets` (with `Status` and `Finalizer` sub-resource)
+- Get/List/Create/Delete/Update/Patch/Watch on `AutoScalingListeners` (with `Status` and `Finalizer` sub-resource)
+- Get/List/Create/Delete/Update/Patch/Watch on `EphemeralRunnerSets` (with `Status` and `Finalizer` sub-resource)
+- Get/List/Create/Delete/Update/Patch/Watch on `EphemeralRunners` (with `Status` and `Finalizer` sub-resource)
+
+- List/Watch on `Pods`
+- List/Watch/Patch on `Roles`
```
Comment (TingluoHuang) [inline docs/adrs/2023-04-11-limit-manager-role-permission.md:67]: the `Patch` is the only change, right?

Comment (nikola-jokic) [reply docs/adrs/2023-04-11-limit-manager-role-permission.md:67]: Right! We can do everything else while holding those resources. It is only the role resource we could not clean up without this extended permission

**Review by TingluoHuang — APPROVED** [review]

**Pull request merged**